cloudvolume/datasource/precomputed/image/common.py

```diff
@@ -68,18 +68,26 @@ def gridpoints(bbox, volume_bbox, chunk_size):
     yield Vec(x,y,z)
 
 def compressed_morton_code(gridpt, grid_size):
+  gridpt = np.asarray(gridpt, dtype=np.uint32)
+  single_input = False
+  if gridpt.ndim == 1:
+    gridpt = np.atleast_2d(gridpt)
+    single_input = True
+
   code = np.uint64(0)
```

Comment:
Now that code is vectorized, the `|=` operator below appears magic. I had to do some experimenting to understand what happens:

```python
>>> import numpy as np
>>> code = np.uint64(0)
>>> bit = np.array([[1,2,3],[4,5,6]], dtype=np.uint64)
>>> bit
array([[1, 2, 3],
       [4, 5, 6]], dtype=uint64)
>>> bit >> np.uint64(1)
array([[0, 1, 1],
       [2, 2, 3]], dtype=uint64)
>>> bit >> np.uint64(1) << np.uint64(1)
array([[0, 2, 2],
       [4, 4, 6]], dtype=uint64)
>>> code | (bit >> np.uint64(1) << np.uint64(1))
array([[0, 2, 2],
       [4, 4, 6]], dtype=uint64)
>>> code | (bit[:,0] >> np.uint64(1) << np.uint64(1))
array([0, 4], dtype=uint64)
```

Reply:
This could be fixed by assigning code to be an array from the start instead of a scalar.